eos-tech-support/eos-diagnostics

```diff
@@ -568,15 +584,90 @@ function dumpDiagnostics(filename) {
     }
 }
 
-let fileArg = ARGV[0];
+function copyAcpiFirmware(output)
+{
+    let dir = Gio.File.new_for_path("/sys/firmware/acpi/tables");
+    let fileEnum;
+
+    output = Gio.File.new_for_path(output);
+
+    try {
+	fileEnum = dir.enumerate_children('standard::name',
+	                                  Gio.FileQueryInfoFlags.NONE, null);
+    } catch (e) {
+	return;
+    }
+
+    let info;
+    while ((info = fileEnum.next_file(null))) {
+	let name = info.get_name();
+	if (!name.startsWith("DSDT") && !name.startsWith("SSDT"))
+	    continue;
+
+	let child = dir.resolve_relative_path(name);
+	let target = output.resolve_relative_path('acpi_' + name.toLowerCase());
+	child.copy(target, 0, null, null);
+    }
+}
+
+function dumpHardwareInfo(filename) {
+    let dir = "/tmp/" + filename;
+    GLib.mkdir_with_parents(dir, parseInt("0700", 8));
+
+    copyAcpiFirmware(dir);
+
+    for (var i = 0; i < diagnostics.length; i++) {
+        let diag = diagnostics[i];
+        if (!diag['hardwareInfo'])
+            continue;
+
+        let name = diag['title'].toLowerCase().replace(' ', '_') + '.txt';
+
+        let content = diag['content']();
+        if (content === null || content == '')
+	    continue;
+
+        content = content.toString().trim() + '\n';
+        GLib.file_set_contents(dir + '/' + name, content);
+    }
+
+    filename += '.tar.xz';
+    let command = 'tar -C ' + dir + ' -cJf ' + filename + ' . ';
+    let [res, stdout, stderr, exitStatus] = GLib.spawn_command_line_sync(command);
+    GLib.spawn_command_line_sync("rm -rf " + dir);
+    if (exitStatus != 0) {
+	print("ERROR: Failed to create archive");
+	return false;
+    }
+
+    print("Wrote " + filename);
+    return true;
+}
+
 let filename;
+if (ARGV[0] == "-h" || ARGV[0] == "--hardware") {
+    filename = ARGV[1];
+
+    if (GLib.get_user_name() != "root") {
+	print("ERROR: For hardware info, run eos-diagnostics as root.");
+	System.exit(1);
+    }
+
+    if (!filename) {
+	let date = GLib.DateTime.new_now_local();
+	filename = 'eos-hwinfo-' + date.format('%y%m%d_%H%M%S_UTC%z');
```

Comment:
It would get saved in root's home directory. I think its best to make it save in the current directory so that you can just open a terminal and do "sudo eos-diagnostics" and then have access to the resultant file in the file browser.